lib/iris/cube.py

```diff
@@ -1599,11 +1599,11 @@ def cell_methods(self, cell_methods):
 
     def core_data(self):
         """
-        The data at the core of this cube.
-        May be a numpy array or a dask array.
-        In using this, you are buying into not caring about the
-        type of the result
-        to be decided: should this be public??
+        Retrieve the data array of this :class:`~iris.cube.Cube` in its
+        current state, which will be either real or lazy.
+
+        If this :class:`~iris.cube.Cube` has lazy data, accessing its data
```

Comment:
I don't like that we lost the purpose description : i.e. it needs saying that the *point* of core_data is :
"so you can implement operations that work irrespective of the type of underlying data, and thus preserve laziness".
I mean of course, something like that but much shorter clearer and more intelligible :  Over to you :wink: